### PR TITLE
[Xtensa] Implement sextload i8

### PR DESCRIPTION
**Pull request by andreisfr**

### CODE REVIEW

**Comment by andreisfr** [comment]
@arsenm , @s-barannikov , please review this PR whenever you have the time.

File: llvm/test/CodeGen/Xtensa/load.ll [hunk 1]
```diff
@@ -0,0 +1,13 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc --mtriple=xtensa < %s | FileCheck %s
+
+define signext i8 @test_load_i8(i32 noundef %0){
```
Comment (s-barannikov) [inline llvm/test/CodeGen/Xtensa/load.ll:4]: Why don't pass the pointer as an argument?
```suggestion
define signext i8 @test_load_i8(ptr %p){
```

Comment (andreisfr) [reply llvm/test/CodeGen/Xtensa/load.ll]: Thank you for comment. Fixed

File: llvm/lib/Target/Xtensa/XtensaISelLowering.cpp [hunk 2]
```diff
@@ -70,11 +70,12 @@ XtensaTargetLowering::XtensaTargetLowering(const TargetMachine &TM,
   setOperationAction(ISD::FP_TO_UINT, MVT::i32, Expand);
   setOperationAction(ISD::FP_TO_SINT, MVT::i32, Expand);
 
-  // No sign extend instructions for i1
+  // No sign extend instructions for i1 and sign extend load i8
```
Comment (s-barannikov) [inline llvm/lib/Target/Xtensa/XtensaISelLowering.cpp:73]: Comments should end with a period.

File: llvm/lib/Target/Xtensa/XtensaISelLowering.cpp [hunk 3]
```diff
@@ -70,11 +70,12 @@ XtensaTargetLowering::XtensaTargetLowering(const TargetMachine &TM,
...
   for (MVT VT : MVT::integer_valuetypes()) {
```
Comment (s-barannikov) [inline llvm/lib/Target/Xtensa/XtensaISelLowering.cpp:74]: FWIW you only need to set actions for legal VTs, that is only for `VT == MVT::i32`.

**Review by s-barannikov — APPROVED** [review]

**Pull request merged**